# SPDX-License-Identifier: GPL-2.0-or-later
#
# Derived from the publicly available Ingenic SDK Buildroot target support:
# https://github.com/Llixuma/ingenic-linux-kernel6.6-x2000-v1.0-20250221
# Ingenic SDK commit: a98c2e1f22e4263ddd4153a4eca4db4dcfd2777b
# Rebase target: upstream Buildroot 2025.02.17
# Upstream commit: d0820dd09916edcefc44e525355afbea30d5bee4
diff --git a/arch/Config.in.mips b/arch/Config.in.mips
index 29b4180d21..2715f0a1c3 100644
--- a/arch/Config.in.mips
+++ b/arch/Config.in.mips
@@ -97,6 +97,11 @@ config BR2_mips_xburst
 	  to be used in order to prevent emitting these instructions.
 
 	  See http://www.ingenic.com/en/?xburst.html
+config BR2_mips_xburst2
+	bool "XBurst2"
+	depends on !BR2_ARCH_IS_64
+	select BR2_MIPS_CPU_MIPS32R5
+	select BR2_MIPS_NAN_2008
 config BR2_mips4
 	bool "MIPS IV"
 config BR2_mips_64
@@ -255,6 +260,7 @@ config BR2_GCC_TARGET_ARCH
 	default "m6201"		if BR2_mips_m6250
 	default "p5600"		if BR2_mips_p5600
 	default "mips32r2"	if BR2_mips_xburst
+	default "mips32r2"	if BR2_mips_xburst2
 	default "mips64"	if BR2_mips_64
 	default "mips64r2"	if BR2_mips_64r2
 	default "mips64r3"	if BR2_mips_64r3
